### ISSUE TITLE
 Unable to order the ISO/templates in the UI 

### ISSUE THREAD

**Issue by kiranchavala** [apache/cloudstack#7278]
##### ISSUE TYPE

 * Bug report

##### COMPONENT NAME

~~~
Componet:UI
Componet:api
~~~

##### CLOUDSTACK VERSION

~~~
4.18 nightly
4.17.2

~~~

##### SUMMARY

Unable to order the ISO/templates in the UI 

Steps to reproduce the issue 

1. Login as admin user 

2. Navigate to Images> Iso

3. Register a sample iso (as a admin user)

4. Try to move the Order the registerd iso 

Move to the top
Move to the bottom

Move up one row 
Move down one row

Similarly for Images> Templates

1. It works partially 

Api calls

https://cloudstack.apache.org/api/apidocs-4.17/apis/updateIso.html

In the backend updateIso api call is executed with sorkey

It updates the database table vm_template.sort_key

~~~

MariaDB [cloud]> select name,sort_key from vm_template where format='iso' ;
+---------------------------------+----------+
| name                            | sort_key |
+---------------------------------+----------+
| vmware-tools.iso                |        5 |
| xs-tools.iso                    |        3 |
| v1.24.0-Kubernetes-Binaries-ISO |        6 |
| v1.23.3-Kubernetes-Binaries-ISO |        4 |
| v1.22.6-Kubernetes-Binaries-ISO |        2 |
| rocky                           |        7 |
+---------------------------------+----------+
6 rows in set (0.00 sec


~~~

##### EXPECTED RESULTS

~~~
Admin User should be able to order the iso's and template based on his requirement 

~~~

##### ACTUAL RESULTS

~~~
user is unable to order the iso's and template based on his requirement 
~~~

![iso](https://user-images.githubusercontent.com/1401014/220628894-0dde628e-3b5c-405e-806d-58f69abdb5c4.png)

**Comment by kiranchavala** [comment]
@borisstoyanov please add the tags for this issue

**Comment by borisstoyanov** [comment]
@kiranchavala are you sure this is an API issue, on first glance it seems like ui/display issue (might be wrong) ?

**Comment by kiranchavala** [comment]
@borisstoyanov it could be API issue as 2 templates/iso can have the same sort_key field in the database, which can lead into issues

**Comment by davidjumani** [comment]
Unrelated to the UI as the API call is correctly made but the results are not sorted

**Comment by CharlesQueiroz** [comment]
Guys, I reproduced the problem, and the ordering is not being maintained. I discovered that the issue is due to using a Hashtable instead of a LinkedHashMap when returning the results to the front end.

As Hashtable does not maintain the insertion order, it returns the list similarly. I've created a PR that solves this problem. 

Here: https://github.com/apache/cloudstack/pull/7844

If anyone could take a look, I'd appreciate it.
